[WTR][Skia] Add support for web contents pixel results

carlosgcampos opened the pull request:
#### bcddb3eea6b4257f6a974f9f78efdfedf9e68c8f
<pre>
[WTR][Skia] Add support for web contents pixel results
<a href="https://bugs.webkit.org/show_bug.cgi?id=271239">https://bugs.webkit.org/show_bug.cgi?id=271239</a>

Reviewed by Alejandro G. Castro.

Add WKImageSkia with WKImageCreateSkImage() to be used from
dumpPixelsAndCompareWithExpected().

* Source/WebKit/Shared/API/c/cairo/WKImageCairo.cpp:
(WKImageCreateCairoSurface):
(WKImageCreateFromCairoSurface):
* Source/WebKit/Shared/API/c/cairo/WKImageCairo.h:
* Source/WebKit/Shared/API/c/skia/WKImageSkia.cpp: Copied from Source/WebKit/Shared/API/c/cairo/WKImageCairo.h.
(WKImageCreateSkImage):
* Source/WebKit/Shared/API/c/skia/WKImageSkia.h: Copied from Source/WebKit/Shared/API/c/cairo/WKImageCairo.h.
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Tools/WebKitTestRunner/gtk/PlatformWebViewGtk.cpp:
* Tools/WebKitTestRunner/skia/TestInvocationSkia.cpp:
(WTR::TestInvocation::dumpPixelsAndCompareWithExpected):

Canonical link: <a href="https://commits.webkit.org/276401@main">https://commits.webkit.org/276401@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/239ccf09e41493fa4d6fe97638b12f3091a38c9b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44570 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23647 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/47018 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47222 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40568 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46875 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27673 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21039 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/36645 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45146 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20706 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38365 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17719 "Found 1 new API test failure: /WPE/TestResources:/webkit/WebKitWebView/sync-request-on-max-conns (failure)") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/18137 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39510 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2616 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40769 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39786 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48850 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19527 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16061 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/43591 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20858 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42347 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/21186 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6132 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20524 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alexgcastro reviewed (approved):
Nice one! Closer to have the testing complete with skia.

carlosgcampos committed:
[WTR][Skia] Add support for web contents pixel results
https://bugs.webkit.org/show_bug.cgi?id=271239

Reviewed by Alejandro G. Castro.

Add WKImageSkia with WKImageCreateSkImage() to be used from
dumpPixelsAndCompareWithExpected().

* Source/WebKit/Shared/API/c/cairo/WKImageCairo.cpp:
(WKImageCreateCairoSurface):
(WKImageCreateFromCairoSurface):
* Source/WebKit/Shared/API/c/cairo/WKImageCairo.h:
* Source/WebKit/Shared/API/c/skia/WKImageSkia.cpp: Copied from Source/WebKit/Shared/API/c/cairo/WKImageCairo.h.
(WKImageCreateSkImage):
* Source/WebKit/Shared/API/c/skia/WKImageSkia.h: Copied from Source/WebKit/Shared/API/c/cairo/WKImageCairo.h.
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Tools/WebKitTestRunner/gtk/PlatformWebViewGtk.cpp:
* Tools/WebKitTestRunner/skia/TestInvocationSkia.cpp:
(WTR::TestInvocation::dumpPixelsAndCompareWithExpected):

Canonical link: https://commits.webkit.org/276401@main
